File: cdc-event-streaming/kafka-streams-consumer/src/main/java/com/onlyteo/sandbox/config/AppKafkaProperties.java
package com.onlyteo.sandbox.config;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "app.kafka")
public record AppKafkaProperties(@NotNull AppKafkaStreamsProperties streams) {

    public record AppKafkaStreamsProperties(@NotBlank String accountSourceTopic,
                                            @NotBlank String statusSourceTopic) {
    }
}

File: spring-boot-cdc-event-streaming/spring-boot-cdc-event-streaming-kafka-streams-consumer/src/main/kotlin/com/onlyteo/sandbox/properties/AppKafkaProperties.kt
package com.onlyteo.sandbox.properties

import jakarta.validation.Valid
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "app.kafka")
data class AppKafkaProperties(@field:Valid @field:NotNull val streams: AppKafkaStreamsProperties)

data class AppKafkaStreamsProperties(
    @field:NotBlank val accountSourceTopic: String,
    @field:NotBlank val statusSourceTopic: String
)
